src/components/application_manager/include/application_manager/application_impl.h

```diff
@@ -65,6 +65,17 @@ using namespace timer;
 namespace mobile_api = mobile_apis;
 namespace custom_str = custom_string;
 
+/**
+ * @brief SwitchApplicationParameters updates application internal parameters
+ * on transport switch. Must be used only for switching flow.
+ * @param app Pointer to switched application
+ * @param app_id New application id (connection key)
+ * @param device_id New device id
+ */
+void SwitchApplicationParameters(ApplicationSharedPtr app,
```

Comment:
@dev-gh why this function is not member function?

Reply:
@LuxoftAKutsan 
Idea was to emphasize that changing core application parameters is not a regular action but should be used only in one specific case. If this method will be put into AM interface I suppose end user may use it for other purposes and that will lead to major problems.